app-backend/database/src/main/scala/com/azavea/rf/database/tables/Projects.scala

```diff
@@ -263,9 +268,53 @@ object Projects extends TableQuery(tag => new Projects(tag)) with LazyLogging {
     database.db.run {
       sceneProjectJoinQuery.result
     } flatMap { alreadyAdded =>
-      val newScenes = sceneIds.filterNot(alreadyAdded.toSet)
+      val newSceneIds = sceneIds.filterNot(alreadyAdded.toSet)
+      val newScenesFootprints = Scenes.getScenesFootprints(sceneIds)
+      val projectAction = Projects.filter(_.id === projectId).result.headOption
+
+      database.db.run(
+        for {
+          footprints <- newScenesFootprints
+          projectOption <- projectAction
+        } yield (footprints, projectOption)
+      ) map { p: (Seq[Option[Projected[Geometry]]], Option[Project]) =>
+        (p._1, p._2) match {
+          case (footprints, Some(project)) => {
+            val newScenesExtent:Option[Extent] = footprints.length match {
+              case 0 => None
+              case _ => {
+                val footprintEnvelopes = footprints map { footprints => footprints } flatMap { _.map { _.geom.envelope } }
+                footprintEnvelopes.length match {
+                  case 0 => None
+                  case _ => Some(footprintEnvelopes reduce { _ combine _ })
+                }
+              }
+            }
+
+            val newProjectExtent:Extent = (project.extent, newScenesExtent) match {
+              case (Some(projectExtent), Some(scenesExtent)) => projectExtent.envelope.combine(scenesExtent)
+              case (Some(projectExtent), _) => projectExtent.geom.envelope
```

Comment:
So if we fail to get a new scenes extent (all scenes removed from the project, e.g.), the project extent is unchanged?